earth_enterprise/src/fusion/tools/gee_test/gee_test.py

```diff
@@ -179,39 +200,146 @@ def WaitForTutorialBuilds():
+  """Gets publish gedb/mapdb path for list of databases.
+
+  Args:
+    dbs: a list of database names/paths.
+
+  Returns:
+    a list of publish gedb/mapdb paths.
+  """
+  publish_gedb_paths = []
+  publish_dbs_base_path = os.path.join(PUBLISHED_DBS_STREAM_SPACE_DIR, HOSTNAME)
+  for db_name in dbs:
+    if db_name and os.path.isabs(db_name):
+      gedb_path = db_name
+    else:
+      gedb_path = GetGedbPath(db_name)
+
+    publish_gedb_paths.append("{}{}".format(publish_dbs_base_path, gedb_path))
+
+  return publish_gedb_paths
+
+
+def PushAndPublishDbs(dbs):
+  """Push and publish list of databases.
+
+  Args:
+    dbs: list of databases.
+
+  Raises:
+    Exception: when unable to find database.
+  """
   # TODO: Be smarter about things that are already pushed and
   #  published. Right now, publishing in this way with unpublish if there is
   #  something already using the target.
   for db in dbs:
-    result = utils.ExecuteCmd(
-        "sudo /opt/google/bin/gequery Databases/%s --outfiles" % db)
-    db_path = ""
-    for line in result.split("\n"):
-      if line.endswith("/gedb"):
-        db_path = line
-        break
-      elif line.endswith("/mapdb"):
-        db_path = line
-        break
+    db_path = GetGedbPath(db)
     if not db_path:
       raise Exception("Unable to find built database %s." % db)
-    print "push and publish", db_path
-    utils.ExecuteCmd("sudo /opt/google/bin/geserveradmin --adddb %s" % db_path)
-    utils.ExecuteCmd("sudo /opt/google/bin/geserveradmin --pushdb %s" % db_path)
+
+    print "push and publish: ", db_path
+    utils.ExecuteCmd("/opt/google/bin/geserveradmin --adddb %s" % db_path)
+    utils.ExecuteCmd("/opt/google/bin/geserveradmin --pushdb %s" % db_path)
     # TODO: Use web api.
     utils.ExecuteCmd(
-        "sudo /opt/google/bin/geserveradmin --publishdb %s --targetpath %s" %
+        "/opt/google/bin/geserveradmin --publishdb %s --targetpath %s" %
         (db_path, db))
 
 
-def RunTests(name):
-  """Run all .py files in the current directory as tests."""
+def RunGeindexcheck(db_name):
+  """Runs geindexcheck for specified database.
+
+  Args:
+    db_name: database name/path.
+  """
+
+  if db_name and os.path.isabs(db_name):
+    db_path = os.path.normpath(db_name)
+  else:
+    db_path = os.path.join("Databases", db_name)
+
+  result = utils.ExecuteCmd(
+      "/opt/google/bin/geindexcheck --mode all --database %s" % db_path,
+      do_log=False, err2out=True)
+
+  if "Fusion Fatal:" in result:
+    print "geindexcheck FAILED for database: ", db_path
+    print "Result: ", result
+  else:
+    print "SUCCESS"
+
+
+def RunGeindexcheckForDbs(dbs):
+  """Runs geindexcheck for list of databases.
+
+  Args:
+    dbs: list of database names/paths.
+  """
+  for db_name in dbs:
+    RunGeindexcheck(db_name)
+
+
+def GetTests(base_dir):
+  """Get tests from cns and unpack them."""
+  utils.ChDir(base_dir)
+  test_sets = [
+      "/cns/ih-d/home/bpeterson/gee_tests/gold_tests.tgz"
```

Comment:
This line references somebody's username.  I'm fine with it for now if it works, but we should look at getting rid of it in the future.